### PR TITLE
Update user-agents and enhance device test

### PR DESCRIPTION
**Pull request by zephraph**
Being as this project hasn't been updated in a bit, the user agents list is a little out of date. I updated the list, made the parser a little more flexible, and update the test to make it easier to debug. 

Instead of having a single device test every user-agent entry is treated as its own test. When the test fails you can see from the error message which entry it was without having to resort to digging through the json file. 

There are failures. I'm trying to work through how to fix those.

### CODE REVIEW

**Comment by alekseykulikov** [comment]
Hi, thank you for the contribution!

The reason for one test: it doesn’t distract from other important parts of the library. You still customize the output and see which UA failed.

I would keep the list of UAs smaller, like 2 or 5 pages, after second pages there are less than 0,1% browsers, and the possibility of failure is small with no reason to complicate a simple algorithm.

**Comment by alekseykulikov** [comment]
Let's simplify the PR: use 2 or 5 pages and just update the json file with UAs and ensure it still works. Thanks!

**Comment by zephraph** [comment]
I updated the device check to pass the tests... there's a few device specific tests in there just because there wasn't anything generic enough to hook off of.

File: test/device.js [hunk 1]
```diff
@@ -8,18 +8,31 @@ const uaIgnore = [
...
-test('user-agents', t => {
-  Object.keys(userAgents).forEach(type => {
-    const expectedType = type === 'computer' ? 'desktop' : type
-    userAgents[type].forEach(device => {
-      if (uaIgnore.includes(device.ua)) return
-      t.is(getDeviceType(device.ua), expectedType)
+Object.keys(userAgents).forEach(type => {
+  const expectedType = type === 'computer' ? 'desktop' : type
+  userAgents[type].forEach((device, index) => {
+    if (uaIgnore.some(ua => !!device.ua.match(ua))) return
```
Comment (zephraph) [inline test/device.js]: I updated the ignore logic to accept regexes to cover more broad cases

File: test/device.js [hunk 2]
```diff
@@ -8,18 +8,31 @@ const uaIgnore = [
   'Mozilla/5.0 (PlayBook; U; RIM Tablet OS 2.1.0; en-US) AppleWebKit/536.2+ (KHTML, like Gecko) Version/7.2.1.0 Safari/536.2+',
   'Mozilla/5.0 (Linux; U; Android 2.2; en-gb; GT-P1000 Build/FROYO) AppleWebKit/533.1 (KHTML, like Gecko) Version/4.0 Mobile Safari/533.1',
   'Mozilla/5.0 (hp-tablet; Linux; hpwOS/3.0.5; U; en-US) AppleWebKit/534.6 (KHTML, like Gecko) wOSBrowser/234.83 Safari/534.6 TouchPad/1.0',
-  'gsa-crawler (Enterprise; T5-JJQHLGRCX9WKD; ssc-aps-notification-docubridge@hc-sc.gc.ca)',
+  /gsa-crawler/,
   'UCWEB/2.0 (Java; U; MIDP-2.0; Nokia203/20.37) U2/1.0.0 UCBrowser/8.7.0.218 U2/1.0.0 Mobile',
   'Dorado WAP-Browser/1.0.0/powerplay/2',
-  'SCH-U365/1.0 NetFront/3.0.22.2.23 (GUI) MMP/2.0'
+  'SCH-U365/1.0 NetFront/3.0.22.2.23 (GUI) MMP/2.0',
+  /vivo\s+\w+\s+Build/
```
Comment (zephraph) [inline test/device.js]: The site reports that these vivo devices should be tablets, but the user agent strings that this pattern matches are [actually phones](http://www.vivo.com/en/). Just googling the model pulls that up. I think they got miscategorized to target the [vivo tablet](https://www.asus.com/us/Tablets/Vivo-Tab-Products/)

**Comment by alekseykulikov** [comment]
Hey, I think, it's important to find the balance in the quality of UA parsing. UA is a mess. There're some devices like Vivo. Let's focus on UX metrics instead and keep it simple :). Just 2 pages, all ignored UAs are in a list (so it's easy to see, how much and what are ignored). I used your expectation logic and updated the test list.
Thank you for the contribution!

**Pull request closed without merging**